java.lang.RuntimeException: publish nacos metadata failed

版本号
dubbo.version 3.1.7
spring-boot.version 2.3.12.RELEASE
spring-cloud.version Hoxton.SR12
spring-cloud-alibaba.version 2.2.10-RC1
nacos-client version 2.2.0
nacos-server version 2.2.0
nacos-all 通过SPI机制注入数据源:postgresql

依赖
spring-cloud-starter-alibaba-nacos-discovery
dubbo-spring-boot-starter

问题点 1：若不配置 use-as-config-center: false use-as-metadata-center: false 则报错标题的异常
问题点 2：若不使用postgresql数据库，使用mysql或derby 不配置use-as-config-center、 use-as-metadata-center可正常启动，
postgresql 扩展数据源插件代码不存在问题已排除数据库源插件代码疑问点
问题点3： 我理解无论是那些数据源mysql、postgresql 对于Dubbo而言都会默认将注册中心的实例同时作为配置中心和元数据中心，但不知道为什么postgresql使用默认配置 use-as-config-center: true use-as-metadata-center: true 就报错

yml配置
spring:
application:
name: rpc
server:
port: 9001
dubbo:
application:
name: rpc
registry:
address: nacos://localhost:8848?username=nacos&password=nacos
protocol:
name: dubbo
port: -1

nacos配置为默认配置

server.servlet.contextPath=/nacos
server.error.include-message=ALWAYS
server.port=8848
spring.datasource.platform=postgresql
db.num=1
db.url.0=jdbc:postgresql://127.0.0.1:5432/nacos_config?reWriteBatchedInserts=true&useUnicode=true&characterEncoding=utf8&serverTimezone=Asia/Shanghai&connectTimeout=1000&socketTimeout=3000&autoReconnect=true&useUnicode=true&useSSL=false
db.user.0=postgres
db.password.0=123456
db.pool.config.driverClassName=org.postgresql.Driver
db.pool.config.connectionTimeout=30000
db.pool.config.validationTimeout=10000
db.pool.config.maximumPoolSize=20
db.pool.config.minimumIdle=2
management.metrics.export.elastic.enabled=false
management.metrics.export.influx.enabled=false
server.tomcat.accesslog.enabled=true
server.tomcat.accesslog.pattern=%h %l %u %t "%r" %s %b %D %{User-Agent}i %{Request-Source}i
server.tomcat.basedir=file:.
nacos.security.ignore.urls=/,/error,//*.css,//.js,/**/.html,//*.map,//.svg,/**/.png,//*.ico,/console-ui/public/,/v1/auth/,/v1/console/health/,/actuator/,/v1/console/server/
nacos.core.auth.system.type=nacos
nacos.core.auth.enabled=false
nacos.core.auth.caching.enabled=true
nacos.core.auth.enable.userAgentAuthWhite=false
nacos.core.auth.server.identity.key=serverIdentity
nacos.core.auth.server.identity.value=security
nacos.core.auth.plugin.nacos.token.expire.seconds=18000
nacos.core.auth.plugin.nacos.token.secret.key=SecretKey012345678901234567890123456789012345678901234567890123456789
nacos.istio.mcp.server.enabled=false

异常信息：

:: Dubbo (v3.1.7) : https://github.com/apache/dubbo
:: Discuss group : dev@dubbo.apache.org
, dubbo version: 3.1.7, current host: 172.0.4.184

. ____ _ __ _ _
/\ / ' __ _ () __ __ _ \ \ \
( ( )__ | '_ | '| | ' / ` | \ \ \
\/ )| |)| | | | | || (| | ) ) ) )
' || .__|| ||| |_, | / / / /
=========||==============|/=////
:: Spring Boot :: (v2.3.12.RELEASE)

2023-03-05 12:08:42.555 INFO 5284 --- [ main] c.b.b.b.c.m.rpc.BossBootRpcApplication : No active profile set, falling back to default profiles: default
2023-03-05 12:08:43.279 INFO 5284 --- [ main] o.apache.dubbo.rpc.model.FrameworkModel : [DUBBO] Dubbo Framework[1] is created, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.300 INFO 5284 --- [ main] o.a.d.c.r.GlobalResourcesRepository : [DUBBO] Creating global shared handler ..., dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.373 INFO 5284 --- [ main] o.a.dubbo.rpc.model.ApplicationModel : [DUBBO] Dubbo Application1.0 is created, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.373 INFO 5284 --- [ main] org.apache.dubbo.rpc.model.ScopeModel : [DUBBO] Dubbo Module[1.0.0] is created, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.430 INFO 5284 --- [ main] o.a.d.c.context.AbstractConfigManager : [DUBBO] Config settings: {dubbo.config.mode=STRICT, dubbo.config.ignore-duplicated-interface=false}, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.431 INFO 5284 --- [ main] o.a.d.c.context.AbstractConfigManager : [DUBBO] Config settings: {dubbo.config.mode=STRICT, dubbo.config.ignore-duplicated-interface=false}, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.447 INFO 5284 --- [ main] o.a.d.c.utils.SerializeSecurityManager : [DUBBO] Serialize check serializable: true, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.448 INFO 5284 --- [ main] o.a.d.c.u.SerializeSecurityConfigurator : [DUBBO] Read serialize allow list from jar:file:/C:/Users/Administrator/.m2/repository/org/apache/dubbo/dubbo/3.1.7/dubbo-3.1.7.jar!/security/serialize.allowlist, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.475 INFO 5284 --- [ main] o.a.d.c.u.SerializeSecurityConfigurator : [DUBBO] Read serialize blocked list from jar:file:/C:/Users/Administrator/.m2/repository/org/apache/dubbo/dubbo/3.1.7/dubbo-3.1.7.jar!/security/serialize.blockedlist, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.625 INFO 5284 --- [ main] o.a.dubbo.rpc.model.ApplicationModel : [DUBBO] Dubbo Application1.1 is created, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.626 INFO 5284 --- [ main] org.apache.dubbo.rpc.model.ScopeModel : [DUBBO] Dubbo Module[1.1.0] is created, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.657 INFO 5284 --- [ main] o.a.d.c.context.AbstractConfigManager : [DUBBO] Config settings: {dubbo.config.mode=STRICT, dubbo.config.ignore-duplicated-interface=false}, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.657 INFO 5284 --- [ main] o.a.d.c.context.AbstractConfigManager : [DUBBO] Config settings: {dubbo.config.mode=STRICT, dubbo.config.ignore-duplicated-interface=false}, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.664 INFO 5284 --- [ main] o.a.d.c.u.SerializeSecurityConfigurator : [DUBBO] Read serialize allow list from jar:file:/C:/Users/Administrator/.m2/repository/org/apache/dubbo/dubbo/3.1.7/dubbo-3.1.7.jar!/security/serialize.allowlist, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.665 INFO 5284 --- [ main] o.a.d.c.u.SerializeSecurityConfigurator : [DUBBO] Read serialize blocked list from jar:file:/C:/Users/Administrator/.m2/repository/org/apache/dubbo/dubbo/3.1.7/dubbo-3.1.7.jar!/security/serialize.blockedlist, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.683 INFO 5284 --- [ main] o.a.d.c.s.c.DubboSpringInitializer : [DUBBO] Use default application: Dubbo Application1.1, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.684 INFO 5284 --- [ main] org.apache.dubbo.rpc.model.ScopeModel : [DUBBO] Dubbo Module[1.1.1] is created, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.691 INFO 5284 --- [ main] o.a.d.c.context.AbstractConfigManager : [DUBBO] Config settings: {dubbo.config.mode=STRICT, dubbo.config.ignore-duplicated-interface=false}, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.697 INFO 5284 --- [ main] o.a.d.c.u.SerializeSecurityConfigurator : [DUBBO] Read serialize allow list from jar:file:/C:/Users/Administrator/.m2/repository/org/apache/dubbo/dubbo/3.1.7/dubbo-3.1.7.jar!/security/serialize.allowlist, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.697 INFO 5284 --- [ main] o.a.d.c.u.SerializeSecurityConfigurator : [DUBBO] Read serialize blocked list from jar:file:/C:/Users/Administrator/.m2/repository/org/apache/dubbo/dubbo/3.1.7/dubbo-3.1.7.jar!/security/serialize.blockedlist, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.700 INFO 5284 --- [ main] o.a.d.c.s.c.DubboSpringInitializer : [DUBBO] Use default module model of target application: Dubbo Module[1.1.1], dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:43.700 INFO 5284 --- [ main] o.a.d.c.s.c.DubboSpringInitializer : [DUBBO] Bind Dubbo Module[1.1.1] to spring container: org.springframework.beans.factory.support.DefaultListableBeanFactory@149dd36b, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:44.233 WARN 5284 --- [ main] o.s.boot.actuate.endpoint.EndpointId : Endpoint ID 'service-registry' contains invalid characters, please migrate to a valid format.
2023-03-05 12:08:44.301 INFO 5284 --- [ main] c.s.b.f.a.ServiceAnnotationPostProcessor : [DUBBO] BeanNameGenerator bean can't be found in BeanFactory with name [org.springframework.context.annotation.internalConfigurationBeanNameGenerator], dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:44.301 INFO 5284 --- [ main] c.s.b.f.a.ServiceAnnotationPostProcessor : [DUBBO] BeanNameGenerator will be a instance of org.springframework.context.annotation.AnnotationBeanNameGenerator , it maybe a potential problem on bean name generation., dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:44.307 INFO 5284 --- [ main] c.s.b.f.a.ServiceAnnotationPostProcessor : [DUBBO] Found 1 classes annotated by Dubbo @service under package [xxx.rpc]: [xxx.rpc.DemoServiceImpl], dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:44.328 INFO 5284 --- [ main] c.s.b.f.a.ServiceAnnotationPostProcessor : [DUBBO] Register ServiceBean[ServiceBean:xxx.service.DemoService::]: Root bean: class [org.apache.dubbo.config.spring.ServiceBean]; scope=; abstract=false; lazyInit=null; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=null; factoryMethodName=null; initMethodName=null; destroyMethodName=null, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:44.451 INFO 5284 --- [ main] o.s.cloud.context.scope.GenericScope : BeanFactory id=af851f6e-7934-3c8d-ab9d-41ac1c2cc380
2023-03-05 12:08:44.934 INFO 5284 --- [ main] f.a.ReferenceAnnotationBeanPostProcessor : [DUBBO] class org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor was destroying!, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:45.639 INFO 5284 --- [ main] o.s.b.w.embedded.tomcat.TomcatWebServer : Tomcat initialized with port(s): 9001 (http)
2023-03-05 12:08:45.652 INFO 5284 --- [ main] o.apache.catalina.core.StandardService : Starting service [Tomcat]
2023-03-05 12:08:45.652 INFO 5284 --- [ main] org.apache.catalina.core.StandardEngine : Starting Servlet engine: [Apache Tomcat/9.0.46]
2023-03-05 12:08:45.957 INFO 5284 --- [ main] o.a.c.c.C.[Tomcat].[localhost].[/] : Initializing Spring embedded WebApplicationContext
2023-03-05 12:08:45.957 INFO 5284 --- [ main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 3377 ms
2023-03-05 12:08:46.760 INFO 5284 --- [ main] o.a.d.c.s.c.DubboConfigBeanInitializer : [DUBBO] loading dubbo config beans ..., dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:46.773 INFO 5284 --- [ main] o.a.d.c.s.c.DubboConfigBeanInitializer : [DUBBO] dubbo config beans are loaded., dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:47.059 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] No value is configured in the registry, the DynamicConfigurationFactory extension[name : nacos] supports as the config center, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:47.061 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] The registry[<dubbo:registry address="nacos://localhost:8848?username=nacos&password=nacos" protocol="nacos" port="8848" parameters="org.apache.dubbo.common.url.component.URLParam$URLParamMap@3686435f" />] will be used as the config center, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:47.076 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] use registry as config-center: <dubbo:config-center highestPriority="false" id="config-center-nacos-localhost-8848" address="nacos://localhost:8848?username=nacos&password=nacos" protocol="nacos" port="8848" parameters="{client=null, password=nacos, username=nacos}" />, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:47.218 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.impl.NacosClientAuthServiceImpl success.
2023-03-05 12:08:47.218 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.ram.RamClientAuthServiceImpl success.
2023-03-05 12:08:51.421 WARN 5284 --- [ main] o.a.d.common.config.ConfigurationUtils : [DUBBO] Config center was specified, but no config item found., dubbo version: 3.1.7, current host: 172.0.4.184, error code: 0-12. This may be caused by , go to https://dubbo.apache.org/faq/0/12 to find instructions.
2023-03-05 12:08:51.421 WARN 5284 --- [ main] o.a.d.common.config.ConfigurationUtils : [DUBBO] Config center was specified, but no config item found., dubbo version: 3.1.7, current host: 172.0.4.184, error code: 0-12. This may be caused by , go to https://dubbo.apache.org/faq/0/12 to find instructions.
2023-03-05 12:08:51.447 INFO 5284 --- [ main] o.a.dubbo.config.context.ConfigManager : [DUBBO] The current configurations or effective configurations are as follows:, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.448 INFO 5284 --- [ main] o.a.dubbo.config.context.ConfigManager : [DUBBO] <dubbo:application parameters="{}" name="rpc" qosEnable="true" protocol="dubbo" />, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.448 INFO 5284 --- [ main] o.a.dubbo.config.context.ConfigManager : [DUBBO] <dubbo:protocol port="-1" name="dubbo" />, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.449 INFO 5284 --- [ main] o.a.dubbo.config.context.ConfigManager : [DUBBO] <dubbo:registry address="nacos://localhost:8848?username=nacos&password=nacos" protocol="nacos" port="8848" parameters="org.apache.dubbo.common.url.component.URLParam$URLParamMap@3686435f" />, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.449 INFO 5284 --- [ main] o.a.dubbo.config.context.ConfigManager : [DUBBO] <dubbo:ssl />, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.490 INFO 5284 --- [ main] o.a.d.c.deploy.DefaultModuleDeployer : [DUBBO] Dubbo Module[1.1.0] has been initialized!, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.497 INFO 5284 --- [ main] o.a.d.c.deploy.DefaultModuleDeployer : [DUBBO] Dubbo Module[1.1.1] has been initialized!, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.504 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] No value is configured in the registry, the MetadataReportFactory extension[name : nacos] supports as the metadata center, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.504 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] The registry[<dubbo:registry address="nacos://localhost:8848?username=nacos&password=nacos" protocol="nacos" port="8848" parameters="org.apache.dubbo.common.url.component.URLParam$URLParamMap@3686435f" />] will be used as the metadata center, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.509 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] use registry as metadata-center: <dubbo:metadata-report address="nacos://localhost:8848?username=nacos&password=nacos" protocol="nacos" port="8848" parameters="{password=nacos, client=null, username=nacos}" />, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.668 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.impl.NacosClientAuthServiceImpl success.
2023-03-05 12:08:51.668 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.ram.RamClientAuthServiceImpl success.
2023-03-05 12:08:51.941 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] Dubbo Application1.1 has been initialized!, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:51.992 WARN 5284 --- [ main] c.n.c.sources.URLConfigurationSource : No URLs will be polled as dynamic configuration sources.
2023-03-05 12:08:51.992 INFO 5284 --- [ main] c.n.c.sources.URLConfigurationSource : To enable URLs as dynamic configuration sources, define System property archaius.configurationSource.additionalUrls or make config.properties available on classpath.
2023-03-05 12:08:51.998 WARN 5284 --- [ main] c.n.c.sources.URLConfigurationSource : No URLs will be polled as dynamic configuration sources.
2023-03-05 12:08:51.998 INFO 5284 --- [ main] c.n.c.sources.URLConfigurationSource : To enable URLs as dynamic configuration sources, define System property archaius.configurationSource.additionalUrls or make config.properties available on classpath.
2023-03-05 12:08:52.398 INFO 5284 --- [ main] o.s.s.concurrent.ThreadPoolTaskExecutor : Initializing ExecutorService 'applicationTaskExecutor'
2023-03-05 12:08:56.984 INFO 5284 --- [ main] o.s.b.a.e.web.EndpointLinksResolver : Exposing 2 endpoint(s) beneath base path '/actuator'
2023-03-05 12:08:57.278 INFO 5284 --- [ main] o.s.b.w.embedded.tomcat.TomcatWebServer : Tomcat started on port(s): 9001 (http) with context path ''
2023-03-05 12:08:57.301 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.impl.NacosClientAuthServiceImpl success.
2023-03-05 12:08:57.301 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.ram.RamClientAuthServiceImpl success.
2023-03-05 12:08:57.445 INFO 5284 --- [ main] c.a.c.n.registry.NacosServiceRegistry : nacos registry, DEFAULT_GROUP rpc 172.0.4.184:9001 register finished
2023-03-05 12:08:58.586 INFO 5284 --- [ main] o.a.d.c.deploy.DefaultModuleDeployer : [DUBBO] Dubbo Module[1.1.1] is starting., dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:08:58.587 INFO 5284 --- [ main] o.a.d.c.d.DefaultApplicationDeployer : [DUBBO] Dubbo Application1.1 is starting., dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:00.236 WARN 5284 --- [ main] org.apache.dubbo.config.ServiceConfig : [DUBBO] Use random available port(20880) for protocol dubbo, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 5-8. This may be caused by , go to https://dubbo.apache.org/faq/5/8 to find instructions.
2023-03-05 12:09:02.022 INFO 5284 --- [ main] org.apache.dubbo.qos.server.Server : [DUBBO] qos-server bind localhost:22222, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.026 INFO 5284 --- [ main] org.apache.dubbo.config.ServiceConfig : [DUBBO] Export dubbo service xxx.service.DemoService to local registry url : injvm://127.0.0.1/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&side=provider&timestamp=1677989338696, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.027 INFO 5284 --- [ main] org.apache.dubbo.config.ServiceConfig : [DUBBO] Register dubbo service xxx.service.DemoService url dubbo://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696 to registry localhost:8848, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.227 INFO 5284 --- [ main] o.a.d.remoting.transport.AbstractServer : [DUBBO] Start NettyServer bind /0.0.0.0:20880, export /172.0.4.184:20880, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.317 INFO 5284 --- [ main] o.a.d.r.c.m.store.MetaCacheManager : [DUBBO] Successfully loaded meta cache from file .metadata.rpc.nacos.localhost:8848, entries 0, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.325 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.impl.NacosClientAuthServiceImpl success.
2023-03-05 12:09:02.326 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.ram.RamClientAuthServiceImpl success.
2023-03-05 12:09:02.601 INFO 5284 --- [ main] o.a.dubbo.metadata.MappingCacheManager : [DUBBO] Successfully loaded mapping cache from file .mapping.rpc, entries 0, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.621 INFO 5284 --- [ main] o.a.d.r.c.m.MigrationRuleListener : [DUBBO] Listening for migration rules on dataId rpc.migration, group DUBBO_SERVICEDISCOVERY_MIGRATION, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.640 INFO 5284 --- [ main] org.apache.dubbo.config.ServiceConfig : [DUBBO] Register dubbo service xxx.service.DemoService url dubbo://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696 to registry localhost:8848, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.658 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.impl.NacosClientAuthServiceImpl success.
2023-03-05 12:09:02.658 INFO 5284 --- [ main] c.a.n.p.a.s.c.ClientAuthPluginManager : [ClientAuthPluginManager] Load ClientAuthService com.alibaba.nacos.client.auth.ram.RamClientAuthServiceImpl success.
2023-03-05 12:09:02.890 INFO 5284 --- [ main] o.a.dubbo.registry.nacos.NacosRegistry : [DUBBO] Register: dubbo://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.905 INFO 5284 --- [ main] o.a.dubbo.registry.nacos.NacosRegistry : [DUBBO] Subscribe: provider://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&category=configurators&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.930 WARN 5284 --- [ main] o.a.dubbo.registry.nacos.NacosRegistry : [DUBBO] Ignore empty notify urls for subscribe url provider://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&category=configurators&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 1-4. This may be caused by , go to https://dubbo.apache.org/faq/1/4 to find instructions.
2023-03-05 12:09:02.938 WARN 5284 --- [ main] o.a.dubbo.registry.nacos.NacosRegistry : [DUBBO] Ignore empty notify urls for subscribe url provider://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&category=configurators&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 1-4. This may be caused by , go to https://dubbo.apache.org/faq/1/4 to find instructions.
2023-03-05 12:09:02.953 INFO 5284 --- [ main] o.a.d.m.d.TypeDefinitionBuilder : [DUBBO] Throw classNotFound (com/google/protobuf/GeneratedMessageV3) in class org.apache.dubbo.metadata.definition.protobuf.ProtobufTypeBuilder, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.956 INFO 5284 --- [ main] org.apache.dubbo.config.ServiceConfig : [DUBBO] Try to register interface application mapping for service xxx.service.DemoService, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.956 INFO 5284 --- [Report-thread-1] o.a.d.m.store.nacos.NacosMetadataReport : [DUBBO] store provider metadata. Identifier : org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b; definition: FullServiceDefinition{parameters=org.apache.dubbo.common.url.component.URLParam$URLParamMap@e2721086} ServiceDefinition [canonicalName=xxx.service.DemoService, codeSource=file:/G:/xxx/target/classes/, methods=[MethodDefinition [name=abc, parameterTypes=[], returnType=java.lang.String]]], dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:02.981 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 1. Next retry delay: 98. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.012 ERROR 5284 --- [Report-thread-1] o.a.d.m.store.nacos.NacosMetadataReport : [DUBBO] Failed to put org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b to nacos {"annotations":[],"canonicalName":"xxx.service.DemoService","codeSource":"file:/G:/xxx/target/classes/","methods":[{"annotations":[],"name":"abc","parameterTypes":[],"parameters":[],"returnType":"java.lang.String"}],"parameters":{"pid":"5284","anyhost":"true","interface":"xxx.service.DemoService","side":"provider","application":"rpc","dubbo":"2.0.2","release":"3.1.7","bind.ip":"172.0.4.184","methods":"abc","background":"false","deprecated":"false","dynamic":"true","service-name-mapping":"true","qos.enable":"true","generic":"false","bind.port":"20880","timestamp":"1677989338696"},"types":[{"enums":[],"items":[],"properties":{},"type":"java.lang.String"}],"uniqueId":"xxx.service.DemoService@file:/G:/xxx/target/classes/"}, cause: publish nacos metadata failed, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 1-37. This may be caused by , go to https://dubbo.apache.org/faq/1/37 to find instructions.

java.lang.RuntimeException: publish nacos metadata failed
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.storeMetadata(NacosMetadataReport.java:387) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.doStoreProviderMetadata(NacosMetadataReport.java:224) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.storeProviderMetadataTask(AbstractMetadataReport.java:283) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.lambda$storeProviderMetadata$0(AbstractMetadataReport.java:271) [dubbo-3.1.7.jar:3.1.7]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_331]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_331]
at java.lang.Thread.run(Thread.java:750) ~[na:1.8.0_331]

2023-03-05 12:09:03.013 ERROR 5284 --- [Report-thread-1] o.a.d.m.store.nacos.NacosMetadataReport : [DUBBO] Failed to put provider metadata org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b in FullServiceDefinition{parameters=org.apache.dubbo.common.url.component.URLParam$URLParamMap@e2721086} ServiceDefinition [canonicalName=xxx.service.DemoService, codeSource=file:/G:/xxx/target/classes/, methods=[MethodDefinition [name=abc, parameterTypes=[], returnType=java.lang.String]]], cause: Failed to put org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b to nacos {"annotations":[],"canonicalName":"xxx.service.DemoService","codeSource":"file:/G:/xxx/target/classes/","methods":[{"annotations":[],"name":"abc","parameterTypes":[],"parameters":[],"returnType":"java.lang.String"}],"parameters":{"pid":"5284","anyhost":"true","interface":"xxx.service.DemoService","side":"provider","application":"rpc","dubbo":"2.0.2","release":"3.1.7","bind.ip":"172.0.4.184","methods":"abc","background":"false","deprecated":"false","dynamic":"true","service-name-mapping":"true","qos.enable":"true","generic":"false","bind.port":"20880","timestamp":"1677989338696"},"types":[{"enums":[],"items":[],"properties":{},"type":"java.lang.String"}],"uniqueId":"xxx.service.DemoService@file:/G:/xxx/target/classes/"}, cause: publish nacos metadata failed, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 3-2. This may be caused by , go to https://dubbo.apache.org/faq/3/2 to find instructions.

java.lang.RuntimeException: Failed to put org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b to nacos {"annotations":[],"canonicalName":"xxx.service.DemoService","codeSource":"file:/G:/xxx/target/classes/","methods":[{"annotations":[],"name":"abc","parameterTypes":[],"parameters":[],"returnType":"java.lang.String"}],"parameters":{"pid":"5284","anyhost":"true","interface":"xxx.service.DemoService","side":"provider","application":"rpc","dubbo":"2.0.2","release":"3.1.7","bind.ip":"172.0.4.184","methods":"abc","background":"false","deprecated":"false","dynamic":"true","service-name-mapping":"true","qos.enable":"true","generic":"false","bind.port":"20880","timestamp":"1677989338696"},"types":[{"enums":[],"items":[],"properties":{},"type":"java.lang.String"}],"uniqueId":"xxx.service.DemoService@file:/G:/xxx/target/classes/"}, cause: publish nacos metadata failed
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.storeMetadata(NacosMetadataReport.java:391) ~[dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.doStoreProviderMetadata(NacosMetadataReport.java:224) ~[dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.storeProviderMetadataTask(AbstractMetadataReport.java:283) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.lambda$storeProviderMetadata$0(AbstractMetadataReport.java:271) [dubbo-3.1.7.jar:3.1.7]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_331]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_331]
at java.lang.Thread.run(Thread.java:750) ~[na:1.8.0_331]
Caused by: java.lang.RuntimeException: publish nacos metadata failed
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.storeMetadata(NacosMetadataReport.java:387) ~[dubbo-3.1.7.jar:3.1.7]
... 6 common frames omitted

2023-03-05 12:09:03.182 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 2. Next retry delay: 77. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.375 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 3. Next retry delay: 78. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.460 WARN 5284 --- [tion-0-thread-1] o.a.dubbo.registry.nacos.NacosRegistry : [DUBBO] Ignore empty notify urls for subscribe url provider://172.0.4.184:20880/xxx.service.DemoService?anyhost=true&application=rpc&background=false&bind.ip=172.0.4.184&bind.port=20880&category=configurators&check=false&deprecated=false&dubbo=2.0.2&dynamic=true&generic=false&interface=xxx.service.DemoService&methods=abc&pid=5284&qos.enable=true&release=3.1.7&service-name-mapping=true&side=provider&timestamp=1677989338696, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 1-4. This may be caused by , go to https://dubbo.apache.org/faq/1/4 to find instructions.
2023-03-05 12:09:03.513 INFO 5284 --- [yTimer-thread-1] stractMetadataReport$MetadataReportRetry : [DUBBO] start to retry task for metadata report. retry times:1, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.514 INFO 5284 --- [Report-thread-1] o.a.d.m.store.nacos.NacosMetadataReport : [DUBBO] store provider metadata. Identifier : org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b; definition: FullServiceDefinition{parameters=org.apache.dubbo.common.url.component.URLParam$URLParamMap@e2721086} ServiceDefinition [canonicalName=xxx.service.DemoService, codeSource=file:/G:/xxx/target/classes/, methods=[MethodDefinition [name=abc, parameterTypes=[], returnType=java.lang.String]]], dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.523 ERROR 5284 --- [Report-thread-1] o.a.d.m.store.nacos.NacosMetadataReport : [DUBBO] Failed to put org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b to nacos {"annotations":[],"canonicalName":"xxx.service.DemoService","codeSource":"file:/G:/xxx/target/classes/","methods":[{"annotations":[],"name":"abc","parameterTypes":[],"parameters":[],"returnType":"java.lang.String"}],"parameters":{"pid":"5284","anyhost":"true","interface":"xxx.service.DemoService","side":"provider","application":"rpc","dubbo":"2.0.2","release":"3.1.7","bind.ip":"172.0.4.184","methods":"abc","background":"false","deprecated":"false","dynamic":"true","service-name-mapping":"true","qos.enable":"true","generic":"false","bind.port":"20880","timestamp":"1677989338696"},"types":[{"enums":[],"items":[],"properties":{},"type":"java.lang.String"}],"uniqueId":"xxx.service.DemoService@file:/G:/xxx/target/classes/"}, cause: publish nacos metadata failed, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 1-37. This may be caused by , go to https://dubbo.apache.org/faq/1/37 to find instructions.

java.lang.RuntimeException: publish nacos metadata failed
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.storeMetadata(NacosMetadataReport.java:387) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.doStoreProviderMetadata(NacosMetadataReport.java:224) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.storeProviderMetadataTask(AbstractMetadataReport.java:283) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.lambda$storeProviderMetadata$0(AbstractMetadataReport.java:271) [dubbo-3.1.7.jar:3.1.7]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_331]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_331]
at java.lang.Thread.run(Thread.java:750) ~[na:1.8.0_331]

2023-03-05 12:09:03.524 ERROR 5284 --- [Report-thread-1] o.a.d.m.store.nacos.NacosMetadataReport : [DUBBO] Failed to put provider metadata org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b in FullServiceDefinition{parameters=org.apache.dubbo.common.url.component.URLParam$URLParamMap@e2721086} ServiceDefinition [canonicalName=xxx.service.DemoService, codeSource=file:/G:/xxx/target/classes/, methods=[MethodDefinition [name=abc, parameterTypes=[], returnType=java.lang.String]]], cause: Failed to put org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b to nacos {"annotations":[],"canonicalName":"xxx.service.DemoService","codeSource":"file:/G:/xxx/target/classes/","methods":[{"annotations":[],"name":"abc","parameterTypes":[],"parameters":[],"returnType":"java.lang.String"}],"parameters":{"pid":"5284","anyhost":"true","interface":"xxx.service.DemoService","side":"provider","application":"rpc","dubbo":"2.0.2","release":"3.1.7","bind.ip":"172.0.4.184","methods":"abc","background":"false","deprecated":"false","dynamic":"true","service-name-mapping":"true","qos.enable":"true","generic":"false","bind.port":"20880","timestamp":"1677989338696"},"types":[{"enums":[],"items":[],"properties":{},"type":"java.lang.String"}],"uniqueId":"xxx.service.DemoService@file:/G:/xxx/target/classes/"}, cause: publish nacos metadata failed, dubbo version: 3.1.7, current host: 172.0.4.184, error code: 3-2. This may be caused by , go to https://dubbo.apache.org/faq/3/2 to find instructions.

java.lang.RuntimeException: Failed to put org.apache.dubbo.metadata.report.identifier.MetadataIdentifier@210f382b to nacos {"annotations":[],"canonicalName":"xxx.service.DemoService","codeSource":"file:/G:/xxx/target/classes/","methods":[{"annotations":[],"name":"abc","parameterTypes":[],"parameters":[],"returnType":"java.lang.String"}],"parameters":{"pid":"5284","anyhost":"true","interface":"xxx.service.DemoService","side":"provider","application":"rpc","dubbo":"2.0.2","release":"3.1.7","bind.ip":"172.0.4.184","methods":"abc","background":"false","deprecated":"false","dynamic":"true","service-name-mapping":"true","qos.enable":"true","generic":"false","bind.port":"20880","timestamp":"1677989338696"},"types":[{"enums":[],"items":[],"properties":{},"type":"java.lang.String"}],"uniqueId":"xxx.service.DemoService@file:/G:/xxx/target/classes/"}, cause: publish nacos metadata failed
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.storeMetadata(NacosMetadataReport.java:391) ~[dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.doStoreProviderMetadata(NacosMetadataReport.java:224) ~[dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.storeProviderMetadataTask(AbstractMetadataReport.java:283) [dubbo-3.1.7.jar:3.1.7]
at org.apache.dubbo.metadata.report.support.AbstractMetadataReport.lambda$storeProviderMetadata$0(AbstractMetadataReport.java:271) [dubbo-3.1.7.jar:3.1.7]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_331]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_331]
at java.lang.Thread.run(Thread.java:750) ~[na:1.8.0_331]
Caused by: java.lang.RuntimeException: publish nacos metadata failed
at org.apache.dubbo.metadata.store.nacos.NacosMetadataReport.storeMetadata(NacosMetadataReport.java:387) ~[dubbo-3.1.7.jar:3.1.7]
... 6 common frames omitted

2023-03-05 12:09:03.575 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 4. Next retry delay: 89. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.799 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 5. Next retry delay: 41. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:03.976 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 6. Next retry delay: 83. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:04.177 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 7. Next retry delay: 92. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:04.373 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 8. Next retry delay: 12. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184
2023-03-05 12:09:04.574 INFO 5284 --- [ main] o.a.d.r.c.m.MetadataServiceNameMapping : [DUBBO] Failed to publish service name mapping to metadata center by cas operation. Times: 9. Next retry delay: 21. Service Interface: xxx.service.DemoService. Origin Content: null. Ticket: 0. Excepted context: rpc, dubbo version: 3.1.7, current host: 172.0.4.184

Process finished with exit code -1

--- a/config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalConfigInfoPersistServiceImpl.java
+++ b/config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalConfigInfoPersistServiceImpl.java
@@ -49,6 +49,7 @@ import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.context.annotation.Conditional;
 import org.springframework.dao.DataAccessException;
 import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.DuplicateKeyException;
 import org.springframework.dao.EmptyResultDataAccessException;
 import org.springframework.dao.IncorrectResultSizeDataAccessException;
 import org.springframework.jdbc.CannotGetJdbcConnectionException;
@@ -173,7 +174,7 @@ public class ExternalConfigInfoPersistServiceImpl implements ConfigInfoPersistSe
             Map<String, Object> configAdvanceInfo, boolean notify) {
         try {
             addConfigInfo(srcIp, srcUser, configInfo, time, configAdvanceInfo, notify);
-        } catch (DataIntegrityViolationException ive) { // Unique constraint conflict
+        } catch (DuplicateKeyException ive) { // Unique constraint conflict
             updateConfigInfo(configInfo, srcIp, srcUser, time, configAdvanceInfo, notify);
         }
     }
@@ -190,7 +191,7 @@ public class ExternalConfigInfoPersistServiceImpl implements ConfigInfoPersistSe
         try {
             addConfigInfo(srcIp, srcUser, configInfo, time, configAdvanceInfo, notify);
             return true;
-        } catch (DataIntegrityViolationException ive) { // Unique constraint conflict
+        } catch (DuplicateKeyException ignore) { // Unique constraint conflict
             return updateConfigInfoCas(configInfo, srcIp, srcUser, time, configAdvanceInfo, notify);
         }
     }
@@ -478,6 +479,14 @@ public class ExternalConfigInfoPersistServiceImpl implements ConfigInfoPersistSe
             try {
                 ConfigInfo oldConfigInfo = findConfigInfo(configInfo.getDataId(), configInfo.getGroup(),
                         configInfo.getTenant());
+                if (oldConfigInfo == null) {
+                    if (LogUtil.FATAL_LOG.isErrorEnabled()) {
+                        LogUtil.FATAL_LOG.error("expected config info[dataid:{}, group:{}, tenent:{}] but not found.",
+                                configInfo.getDataId(), configInfo.getGroup(), configInfo.getTenant());
+                    }
+                    return Boolean.FALSE;
+                }
+                
                 String appNameTmp = oldConfigInfo.getAppName();
                 /*
                  If the appName passed by the user is not empty, use the persistent user's appName,
@@ -511,6 +520,13 @@ public class ExternalConfigInfoPersistServiceImpl implements ConfigInfoPersistSe
             try {
                 ConfigInfo oldConfigInfo = findConfigInfo(configInfo.getDataId(), configInfo.getGroup(),
                         configInfo.getTenant());
+                if (oldConfigInfo == null) {
+                    if (LogUtil.FATAL_LOG.isErrorEnabled()) {
+                        LogUtil.FATAL_LOG.error("expected config info[dataid:{}, group:{}, tenent:{}] but not found.",
+                                configInfo.getDataId(), configInfo.getGroup(), configInfo.getTenant());
+                    }
+                    return Boolean.FALSE;
+                }
                 String appNameTmp = oldConfigInfo.getAppName();
                 /*
                  If the appName passed by the user is not empty, use the persistent user's appName,
@@ -725,8 +741,7 @@ public class ExternalConfigInfoPersistServiceImpl implements ConfigInfoPersistSe
             ConfigTagsRelationMapper configTagsRelationMapper = mapperManager.findMapper(
                     dataSourceService.getDataSourceType(), TableConstant.CONFIG_TAGS_RELATION);
             sqlCount = configTagsRelationMapper.findConfigInfo4PageCountRows(paramsMap, tagArr.length);
-            sql = configTagsRelationMapper.findConfigInfo4PageFetchRows(paramsMap, tagArr.length, startRow,
-                    pageSize);
+            sql = configTagsRelationMapper.findConfigInfo4PageFetchRows(paramsMap, tagArr.length, startRow, pageSize);
         } else {
             ConfigInfoMapper configInfoMapper = mapperManager.findMapper(dataSourceService.getDataSourceType(),
                     TableConstant.CONFIG_INFO);

